collectors/0/kafka_jmx.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+
+import os
+import re
+import signal
+import subprocess
+import sys
+import time
+
+from collectors.lib import utils
+
+# How oftent to poll
+INTERVAL="60"
+
+# If this user doesn't exist, we'll exit immediately.
+# If we're running as root, we'll drop privileges using this user.
+USER = "kafka_user"
+
+# We add those files to the classpath if they exist.
+CLASSPATH = [
+    "/usr/lib/jvm/default-java/lib/tools.jar",
+]
+
+# Map certain JVM stats so they are unique and shorter
+JMX_SERVICE_RENAMING = {
+  "GarbageCollector": "gc",
+  "OperatingSystem": "os",
+  "Threading": "threads",
+}
+
+IGNORED_METRICS = frozenset(["Loggers", "MBeanName"])
+
+# How many times, maximum, will we attempt to restart the JMX collector.
+# If we reach this limit, we'll exit with an error.
+MAX_RESTARTS = 10
+
+TOP = False  # Set to True when we want to terminate.
+RETVAL = 0    # Return value set by signal handler.
+
+
+def kill(proc):
+  """Kills the subprocess given in argument."""
+  # Clean up after ourselves.
+  proc.stdout.close()
+  rv = proc.poll()
+  if rv is None:
+      os.kill(proc.pid, 15)
+      rv = proc.poll()
+      if rv is None:
+          os.kill(proc.pid, 9)  # Bang bang!
+          rv = proc.wait()  # This shouldn't block too long.
+  print >>sys.stderr, "warning: proc exited %d" % rv
+  return rv
+
+
+def do_on_signal(signum, func, *args, **kwargs):
+  """Calls func(*args, **kwargs) before exiting when receiving signum."""
+  def signal_shutdown(signum, frame):
+    print >>sys.stderr, "got signal %d, exiting" % signum
+    func(*args, **kwargs)
+    sys.exit(128 + signum)
+  signal.signal(signum, signal_shutdown)
+
+
+def main(argv):
+    utils.drop_privileges(user=USER)
+    # Build the classpath.
+    dir = os.path.dirname(sys.argv[0])
+    jar = os.path.normpath(dir + "/../lib/jmx-1.0.jar")
+    if not os.path.exists(jar):
+        print >>sys.stderr, "WTF?!  Can't run, %s doesn't exist" % jar
+        return 13
+    classpath = [jar]
+    for jar in CLASSPATH:
+        if os.path.exists(jar):
+            classpath.append(jar)
+    classpath = ":".join(classpath)
+
+    jpid = "Kafka"
+    jps = subprocess.check_output("jps").split("\n")
```

Comment:
Unless I'm missing something, this requires Python 2.7, I'm not sure we should merge it. @tsuna?